Integrate open PRs and update instructions

Emilios1995 opened the pull request:
Before the transfer, I made a fork and merged all the PRs that looked good to me. This set of PRs includes the one that refactors the repo so that it's easy to install everything without the separate `nvim-treesitter-rescript`. I also updated the Readme so that the install instructions point to the new location. (I also updated an outdated reference to the `InspectTree` command.)

 The PRs are: #230, #232, #233, #235, #237

After this is merged into main, I'll try to get this parser included in Neovim by default to make it even easier to install. 


### Failing wild tests

These are the missing features that are causing the wild tests to fail:

Failing tests: 

- [x]  **`as` annotations on record fields**.
    
    Source: `/rescript-nodejs.git/src/WorkerThreads.res`
    
    Snippet:
    
    ```reason
    type options<'a, 'env> = {
        argv: array<string>,
        env: {..} as 'env,
        // ...
      }
    ```
    
- [x]  **Optional fields in switch statements (it also fails in deconstruction patterns in `let` bindings)**
    
    Source: `/rescript-lang.org/scripts/gendocs.res:152`
    
    Snippet: 
    
    ```reason
        switch docItem {
        | Value({id, name, docstrings, signature, ?deprecated}) => {
           //...
    ```
    
- [x]  **Partial application**
    
    Sources: 
    
    - `rescript-lang.org/src/Playground.res:1637`
    - `rescript-lang.org/src/SyntaxLookup.res:121`
    - `rescript-lang.org/src/bindings/RescriptCompilerApi.res:247`
    - `rescript-lang.org/src/common/BlogFrontmatter.res:127`
    - `rescript-association/rescript-lang.org/src/common/DocFrontmatter.res:11`
    - `rescript-association/rescript-lang.org/src/vendor/Json_decode.res:198`
    
    Snippet:
    
    ```reason
    Js.Array2.map(locMsgs, locMsgToCmError(~kind=#Error, ...))
    ```

Emilios1995 commented:
Looks like the wild tests are failing. Those are useful, but I don't think it's a good idea to run them as checks on PRs since a change upstream could cause something to fail even if it's completely unrelated to the PR.

aspeddro commented:
https://github.com/rescript-lang/tree-sitter-rescript/pull/233 should parse type paramaters in type spread

An error example from test wild: https://github.com/rescript-lang/tree-sitter-rescript/actions/runs/8234279999/job/22515674801?pr=239#step:8:26

[Code](https://github.com/cca-io/rescript-mui/blob/34a4a79eefcaa4e87d7804e03ef57b6411bed4e8/packages/rescript-mui-material/src/components/FilledInput.res#L41)

```rescript
type props<'value> = {
  ...InputBase.publicProps<'value>
}
```

aspeddro commented:
Since we updated the installation instructions, we should remove `src` from `.gitignore`

Emilios1995 commented:
> Since we updated the installation instructions, we should remove `src` from `.gitignore`

That's probably right. I think we could also ask users to run `TSInstallFromGrammar`, but checking the generated parser in is simpler for sure. But also, I think we should change the `requires_generate_from_grammar` option in the installation instruction snippet to `false`, right?

Emilios1995 commented:
> > Since we updated the installation instructions, we should remove `src` from `.gitignore`
> 
> That's probably right. I think we could also ask users to run `TSInstallFromGrammar`, but checking the generated parser in is simpler for sure. But also, I think we should change the `requires_generate_from_grammar` option in the installation instruction snippet to `false`, right?

@aspeddro I looked it up. We shouldn't need to check in the files in `src` because the snippet in the instructions has `requires_generate_from_grammar = true`. When that option is set, Neovim will build generate the parser regardless of whether the user called `TSInstall` or `TSInstallFromGrammar`. See [here](https://github.com/nvim-treesitter/nvim-treesitter/blob/2014f8285e306f4c8a896c7d3af53fdd599e5a4f/lua/nvim-treesitter/install.lua#L346).

Of course, I'm open to checking in the generated files if you think that'll help with other editors, but this [argument](https://github.com/alex-pinkus/tree-sitter-swift?tab=readme-ov-file#where-is-your-parserc) from the maintainer of the Swift parser for _not_ checking it in makes sense to me. 


--- 

Two more things: 

1. You may have seen that I fixed the type spreads wild test that was failing (if not, see my previous commit.) However, when looking at the tree, I thought it would be much nicer to have separate nodes for type spreads. I made that change in a [PR](https://github.com/Emilios1995/tree-sitter-rescript/pull/1) against _my_ `main`, which is the target of _this_ PR. Please take a look, and if you agree, I'll integrate it here. 
2. I edited the description of this PR with the missing features that are making the tests fail. As we discussed, we can make all the required changes here. If you want to implement some of those, feel free to make a PR against my fork's `main` branch, which is this PR's target.

Emilios1995 commented on test/corpus/jsx.txt:
```diff
@@ -310,3 +310,20 @@ Spread props
+Elements in pipes
+================================================================================
+
+<Comp />->Some
```
A similar snippet was failing in the wild tests (but I forgot to add it to the list in the description.)

The problem was that pipe expressions only accept `primary_expression`s as their members, and JSX elements wasn't one. I fixed it by simply adding it to the list of `primary_expression`s. I looked at the grammar and this shouldn't have any negative repercussions.

aspeddro commented:
I agree with `type_spread` node. It's also great that we don't need to track `src/parser.c`

Emilios1995 commented:
> What do you think about updating the highlights in another PR?

Sounds good, I can do that. 

Regarding this PR, I just merged the `type_spread` change, and all the wild tests are now passing.

aspeddro reviewed (approved):
Awesome work @Emilios1995 🎉
